### ISSUE TITLE
Podman v3 miss all registries except listed in short names

### ISSUE THREAD

**Issue by Friz-zy** [containers/podman#9256]
/kind bug

**Description**

Podman at least version 3 from testing repo comes without `/etc/containers/registries.conf`. As a result it doesn't know anythin about any registry and images except listed in `/etc/containers/registries.conf.d/shortnames.conf`

**Steps to reproduce the issue:**

```
podman pull ubuntu
Error: error getting default registries to try: short-name "ubuntu" did not resolve to an alias and no unqualified-search registries are defined in ""
```

**Output of `podman version`:**

```
podman version 3.0.0-rc1
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.19.2
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: 'conmon: /usr/libexec/podman/conmon'
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.24, commit: '
  cpus: 4
  distribution:
    distribution: ubuntu
    version: "18.04"
  eventLogger: journald
  hostname: zen
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 5.4.0-54-generic
  linkmode: dynamic
  memFree: 9947365376
  memTotal: 25099407360
  ociRuntime:
    name: crun
    package: 'crun: /usr/bin/crun'
    path: /usr/bin/crun
    version: |-
      crun version 0.17.5-86d6-dirty
      commit: fd582c529489c0738e7039cbc036781d1d039014
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: true
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    selinuxEnabled: false
  slirp4netns:
    executable: ""
    package: ""
    version: ""
  swapFree: 5578420224
  swapTotal: 5578420224
  uptime: 74h 17m 34.53s (Approximately 3.08 days)
registries: {}
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 0
    paused: 0
    running: 0
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageStore:
    number: 34
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 3.0.0
  Built: 0
  BuiltTime: Thu Jan  1 03:00:00 1970
  GitCommit: ""
  GoVersion: go1.15.2
  OsArch: linux/amd64
  Version: 3.0.0-rc1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
ii  containernetworking-plugins   0.9.0~3
ii  containers-common                1:1~4
ii  podman                                   3.0.0~0.4.rc1
ii  podman-plugins                       1.1.1~1
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Comment by mheon** [comment]
Which testing repository are you referring to?

**Comment by Friz-zy** [comment]
@mheon `https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/testing/xUbuntu_${VERSION_ID}` from [the install manual](https://podman.io/getting-started/installation)
```
. /etc/os-release
echo "deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/testing/xUbuntu_${VERSION_ID}/ /" | sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:testing.list
curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/testing/xUbuntu_${VERSION_ID}/Release.key | sudo apt-key add -
sudo apt-get update -qq
sudo apt-get -qq -y install podman
```

Stable version (v2) come with `/etc/containers/registries.conf` that contain docker.io and quay.io.
Test packages, however, miss it.

So this looks like a packaging issue, like #9257

**Comment by vrothberg** [comment]
@lsm5 PTAL

**Comment by lsm5** [comment]
Should be fixed in the current kubic testing. Please reopen if issue persists.

**Comment by asottile** [comment]
@lsm5 just checking (because I'm unfamiliar with the terminology) -- should I expect this to be fixed in 3.0.0 or do I wait for a future release for this

Are there any workarounds for this?

**Comment by buck2202** [comment]
This seems to still be an issue in the current stable/testing (which at this moment, show the same version numbers)

```
$ apt-cache madison podman containers-common 
    podman | 100:3.0.0-1 | http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/testing/xUbuntu_18.04  Packages
containers-common |    100:1-7 | http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/testing/xUbuntu_18.04  Packages

$ ls /etc/containers/*
/etc/containers/containers.conf  /etc/containers/policy.json  /etc/containers/storage.conf

/etc/containers/registries.conf.d:
shortnames.conf

/etc/containers/registries.d:
default.yaml

$ cat /etc/containers/policy.json 
{
    "default": [
        {
            "type": "insecureAcceptAnything"
        }
    ],
    "transports":
        {
            "docker-daemon":
                {
                    "": [{"type":"insecureAcceptAnything"}]
                }
        }
}

$ podman pull ubuntu
Error: error getting default registries to try: short-name "ubuntu" did not resolve to an alias and no unqualified-search registries are defined in ""

$ echo -e "[registries.search]\nregistries = ['docker.io']" | sudo tee /etc/containers/registries.conf
[registries.search]
registries = ['docker.io']

$ podman pull ubuntu
Resolving "ubuntu" using unqualified-search registries (/etc/containers/registries.conf)
Trying to pull docker.io/library/ubuntu:latest...
Getting image source signatures
Copying blob f611acd52c6c done  
Copying blob db98fc6f11f0 done  
Copying blob 83ee3a23efb7 done  
Copying config f63181f19b done  
Writing manifest to image destination
Storing signatures
f63181f19b2fe819156dcb068b3b5bc036820bec7014c5f77277cfa341d4cb5e
```
